Plugins/Json/include/Acts/Plugins/Json/GeometryHierarchyMapJsonConverter.hpp

```diff
@@ -102,9 +104,35 @@ class GeometryHierarchyMapJsonConverter {
 
 // implementations
 
-template <typename value_t>
-nlohmann::json GeometryHierarchyMapJsonConverter<value_t>::toJson(
-    const Container& container) const {
+// auxiliary struct to indicate a missing specialization of a template which
+// requires specialisation
+template <typename T, class decorator_t>
+struct missing_specialization {
+  enum { value = false };
+};
+
+// methods to adapt type decorations for the given decorator
+template <class T, class decorator_t>
+void decorateJson([[maybe_unused]] const decorator_t* decorator, [[maybe_unused]] const T& src,
+                  [[maybe_unused]] nlohmann::json& dest) {
+  // this needs to be specialised
+  static_assert(
+      missing_specialization<T, decorator_t>::value,
+      "Explicit specialization needed for each decorator_t and src T");
+  src.compilationFailure();
```

Comment:
Just out of curiosity: What does this line do and why is it only present in this overload and not the other?

Reply:
In principle it could be reduced to static_assert( false, "...error message..."), but then gcc and clang do not mention the template parameters in the error message. So, this construct is there to produce hopefully understandable error messages during compilation, if there is no specialization of decorateJson for the given template parameters T and src.

Reply:
Thanks for explaining! Should ```src.compilationFailure();``` not also be added in l. 131 ?

Reply:
Sorry, I did not read Your first comment correctly. This was the initial idea to provoke a compilation error. This is a remnant I should delete. Thanks !

Reply:
Oh, sorry, I should have been more clear on what I meant :)

Reply:
removed remnant.